src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexWriteOperation.cs

```diff
@@ -206,5 +215,24 @@ protected class IndexWriteOperationStats
             public IndexingStatsScope AddStats;
             public IndexingStatsScope SuggestionStats;
         }
+
+        public byte[] GetBuffer(int necessarySize)
```

Comment:
I think there is an issue here, You are returning an array that may be *larger* than the requested size.
But the calling code is expecting just an array, and may consider all of it to be full.

Reply:
Check my other answer.